Module_3A/hang-man_game.rb

```diff
@@ -0,0 +1,93 @@
+# Putting things into class:
+
+class Game
+    attr_reader :lives,:board,:my_secret_word
+    # -------INIT STATE------------
+    def initialize(word_List)
+        @lives = 7
+        @my_secret_word = word_List.sample
+        @board = set_up_the_board
+    end
+
+    # ---------SETTING UP THE BOARD---------
+    def set_up_the_board
+        Array.new(my_secret_word.size){"_"}
+    end
+
+
+    # ---------BOARD STATE----------
+    def board_state
+        board.join(" ")
+    end
+
+    # --------WIN STATUS------------
+    def won?
+        board.join("") == my_secret_word
+    end
+
+    def lost?
+        lives <= 0
+    end
+
+    # -------- MAKE GUESS -----------
+    def make_guess
+        print "Please guess a letter:"
+        gets.chomp #implicit return occurs here
+    end
+
+    def update_board(guess)
+        if my_secret_word.include?(guess)
+            i = 0
+            while i < my_secret_word.size
+                # if the guess was "A" -> it gives 'a'
+                if guess.downcase == my_secret_word[i].downcase                
+                    board[i] = my_secret_word[i]
+                end
+                i += 1
+            end
+        else
+            @lives -= 1
+        end
+    end
+end
+
+
+# Inheriting from the Game class to create our HangMan Game
+class Hangman < Game
```

Comment:
See if you can create one more game similiar to Hangman using the original Game class

Reply:
This just looks like you modified the previous solution a little bit, so please check if you can create one more game here